src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteTagHelperProviderService.cs

```diff
@@ -44,10 +120,34 @@ internal RemoteTagHelperProviderService(IServiceBroker serviceBroker, ITelemetry
         int lastResultId,
         CancellationToken cancellationToken)
     {
-        var currentTagHelpers = solution.GetProject(projectHandle.ProjectId) is Project workspaceProject
-            ? await _tagHelperResolver.GetTagHelpersAsync(workspaceProject, projectHandle.Configuration, factoryTypeName, cancellationToken).ConfigureAwait(false)
-            : ImmutableArray<TagHelperDescriptor>.Empty;
+        ImmutableArray<Checksum> checksums;
+
+        if (solution.GetProject(projectHandle.ProjectId) is not Project workspaceProject)
+        {
+            checksums = ImmutableArray<Checksum>.Empty;
+        }
+        else
+        {
+            var tagHelpers = await _tagHelperResolver
+                .GetTagHelpersAsync(workspaceProject, projectHandle.Configuration, factoryTypeName, cancellationToken)
+                .ConfigureAwait(false);
+
+            using var _ = ArrayBuilderPool<Checksum>.GetPooledObject(out var builder);
```

Comment:
>ArrayBuilderPool [](http://example.com/codeflow?start=26&length=16)

Could you use PooledArrayBuilder instead as you did above?

Reply:
Sure. Note that the reason I didn't is because `PooledArrayBuilder` is a ref struct and this is an async method. It's easy enough to call a static local method like I did above though.